tests/acl/test_acl_dscp_change.py

```diff
@@ -0,0 +1,654 @@
+#! /usr/bin/env python3
+'''
+    These tests check the Vxlam ecmp nexthop group switch over functionality. Further details are
+    provided with each test.
+'''
+import os
+import time
+import logging
+from collections import defaultdict
+import pytest
+from tests.common.helpers.assertions import pytest_assert as py_assert
+
+from tests.common.vxlan_ecmp_utils import Ecmp_Utils     # noqa F401
+import ptf.testutils as testutils
+from ptf import mask
+from scapy.all import Ether, IP, VXLAN, IPv6, UDP
+
+Logger = logging.getLogger(__name__)
+ecmp_utils = Ecmp_Utils()
+
+
+# This is the list of encapsulations that will be tested in this script.
+SUPPORTED_ENCAP_TYPES = ['v4_in_v4', 'v6_in_v4', 'v4_in_v6', 'v6_in_v6']
+DESTINATION_PREFIX = 150
+NEXTHOP_PREFIX = 100
+pytestmark = [
+    pytest.mark.acl,
+    pytest.mark.disable_loganalyzer,  # Disable automatic loganalyzer, since we use it for the test
+    # This script supports any T1 topology: t1, t1-64-lag, t1-56-lag, t1-lag.
+    pytest.mark.topology("t1"),
+
+]
+
+BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+FILES_DIR = os.path.join(BASE_DIR, "files")
+TEMPLATES_DIR = os.path.join(BASE_DIR, "templates")
+
+TMP_DIR = '/tmp'
+ACL_TABLE_FILE = 'acl_tbl_config.json'
+ACL_TABLE_REMOVE_FILE = "acl_tbl_del.json"
+
+ACL_RULES_FILE = 'acl_rule_config.json'
+ACL_RULES_REMOVE_FILE = "acl_rules_del.json"
+ACL_TEMPLATE_TABLE_FILE = "dscp_acl_tablev4_v6.j2"
+ACL_TEMPLATE_RULE_FILE = "dscp_acl_rulev4_v6.j2"
+
+TABLE_V4 = "OVERLAY_MARK_META_TEST"
+TABLE_V6 = "OVERLAY_MARK_META_TESTV6"
+
+
+@pytest.fixture(
+    name="encap_type",
+    scope="module",
+    params=SUPPORTED_ENCAP_TYPES)
+def fixture_encap_type(request):
+    '''
+        This fixture forces the script to perform one encap_type at a time.
+        So this script doesn't support multiple encap types at the same.
+    '''
+    return request.param
+
+
+@pytest.fixture(scope='module')
+def prepare_test_port(rand_selected_dut, tbinfo):
+    mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
+    dut_port = list(mg_facts['minigraph_portchannels'].keys())[0]
+    if not dut_port:
+        pytest.skip('No portchannels found')
+    if "Ethernet" in dut_port:
```

Comment:
`mg_facts['minigraph_portchannels'].keys()` should only have portchannels. Why do we need this check?